src/twisted/internet/defer.py

```diff
@@ -295,13 +299,55 @@ def __init__(self, canceller=None):
 
         @type canceller: a 1-argument callable which takes a L{Deferred}. The
             return result is ignored.
+
+        @param context: The context that this Deferred should run its callbacks
+        under.
+
+        @type context: L{contextvars.Context}
         """
         self.callbacks = []
         self._canceller = canceller
         if self.debug:
             self._debugInfo = DebugInfo()
             self._debugInfo.creator = traceback.format_stack()[:-1]
 
+        if context is not None:
+            self._context = context
+        else:
+            self._context = _copy_context()
+
+
+    def getContext(self) -> _Context:
+        """
+        Get the context that this Deferred runs its callbacks with.
+
+        @raise L{RuntimeError}: If L{contextvars} is not available.
+
+        @rtype: L{contextvars.Context}
+        """
+        if not _contextvarsSupport:
+            raise RuntimeError(
+                "contextvars support is not available on this platform."
+            )
+
+        return self._context
+
+
+    def setContext(self, context: _Context) -> NoReturn:
```

Comment:
Why does this never return? The `self._context = context` bit runs at the end of the function, after which it implicitly returns (`None`).

Should this maybe be `Optional[NoReturn]`? If that even affects anything.